graphbolt/src/csc_sampling_graph.cc

```diff
@@ -781,10 +783,15 @@ inline int64_t LaborPick(
     std::iota(picked_data_ptr, picked_data_ptr + num_neighbors, offset);
     return num_neighbors;
   }
-  torch::Tensor heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
   // Assuming max_degree of a vertex is <= 4 billion.
-  auto heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
-      heap_tensor.data_ptr<int32_t>());
+  std::array<std::pair<float, uint32_t>, StackSize> heap;
+  auto heap_data = heap.data();
+  torch::Tensor heap_tensor;
+  if (fanout > StackSize) {
+    heap_tensor = torch::empty({fanout * 2}, torch::kInt32);
+    heap_data = reinterpret_cast<std::pair<float, uint32_t>*>(
```

Comment:
Are we assuming the memory size allocated by `torch::empty` equals `fanout * 2 * sizeof(float) `?

Reply:
we are assuming it equals fanout int32_t and fanout floats.